dod/src/main/java/com/revolut/rxdata/dod/DataObservableDelegate.kt

```diff
@@ -98,23 +98,22 @@ class DataObservableDelegate<Params : Any, Domain : Any> constructor(
      * Requests data from network and subscribes to updates
      * (can be triggered by other subscribers or manual cache overrides)
      *
-     * @param forceReload - if true network request will be made even if data exists in caches
+     * @param loadingStrategy - [LoadingStrategy]
      */
-    @Suppress("RedundantLambdaArrow")
-    fun observe(params: Params, forceReload: Boolean = true): Observable<Data<Domain>> =
+    fun observe(params: Params, loadingStrategy: LoadingStrategy): Observable<Data<Domain>> =
         Observable.defer {
             val memCache = fromMemory(params)
             val memoryIsEmpty = memCache == null
             val subject = subject(params)
-            val loading = forceReload || memoryIsEmpty || failedNetworkRequests.containsKey(params)
+            val loading = loadingStrategy.refreshMemory || memoryIsEmpty || failedNetworkRequests.containsKey(params)
```

Comment:
shouldn't it be like
`(loadingStrategy.refreshMemory && (memoryIsEmpty && failedNetworkRequests.containsKey(params))`?

Reply:
I think with that change we can break current behaviour, as at least we should wait for `failedNetworkRequests.containsKey(params)`